### DISCUSSION TITLE
Requesting Dark Theme Variant does not work for FluentTheme

### DISCUSSION THREAD

**Discussion by FrankenApps** [AvaloniaUI/Avalonia#12748] [Q&A] [answered]
**Describe the bug**
When setting the `Application.Current.RequestedThemeVariant` to `ThemeVariant.Dark` nothing changes when the `FluentTheme` is active.

It works fine when the `SimpleTheme` is active, and the `Application.Current.ActualThemeVariant` correctly reports `Dark`.

**To Reproduce**
1. Clone the branch `avalonia-v.0.11` from over [here](https://github.com/FrankenApps/Avalonia-CustomTitleBarTemplate/tree/avalonia-v.0.11).
2. Make sure the **Default Style** toggle button is inactive and then toggle the **Dark Theme** button.
3. The window will stay white and the text on the toggle buttons stays black.

**Expected behavior**
I expected the window to turn dark and the text on the toggle buttons white, just like in the `SimpleTheme`.

Also I expected to actually "see" the toggle buttons in the `FluentTheme`, but this might be another issue.

**Screenshots**
FluentTheme with _Light_ variant:
![Bildschirm­foto 2023-08-04 um 19 45 42](https://github.com/AvaloniaUI/Avalonia/assets/17967303/ce260e87-eba2-474a-9a45-05692b44759b)

FluentTheme with _Dark_ variant (not working as expected):
![Bildschirm­foto 2023-08-04 um 19 45 53](https://github.com/AvaloniaUI/Avalonia/assets/17967303/a769ccfd-4281-4c24-9517-140207a41f38)

**Desktop (please complete the following information):**

- OS: Windows
- Version 10

**Additional context**
Works as expected with v.0.10.

**Comment by timunie** [comment]
toggle dark / light variant should work but swapping out entire theme is a heavy lifting. Check how ControlCatalog does it.

**Comment by timunie** [comment]
Sent you a PR to fix FluentTheme toggle Light / Dark. Swapping entire theme is not possible without opening the window again. Correct me if I'm wrong @maxkatz6

**Comment by maxkatz6** [comment, marked as answer]
@timunie is right, as ControlThemes are statically resolved. And if you try to dynamically change set of these static resources, existing controls won't be refreshed. It is not intended to do so, as swapping different themes in runtime is not a supported scenario.

But theme variant switching, like Dark or Light or other, should work fine. Having both fluent and simple themes included at once would definitely affect this behavior.

> Swapping entire theme is not possible without opening the window again.

Window re-opening is one of ways to do it, yes. Probably the best for this kind of demo apps.
Another option would be to recreate content of the window, or mark or reference Control.Theme with DynamicResource for all of your controls (not really recommended though due to performance).

You also can try to remove any theme includes from Application.xaml, and instead add them to your windows independently.

**Comment by timunie** [comment]
I'm converting this into a discussion as default behavior is by design sadly. But performance also kind of matters 😃

**Reply by timunie** [reply]
We still have a feature request open to improve it here: https://github.com/AvaloniaUI/Avalonia/issues/12457
